src/Mvc/Mvc.Core/src/Formatters/SystemTextJsonOutputFormatter.cs

```diff
@@ -77,7 +77,7 @@ public sealed override async Task WriteResponseBodyAsync(OutputFormatterWriteCon
                 // For polymorphic scenarios where the user declares a return type, but returns a derived type,
                 // we want to serialize all the properties on the derived type. This keeps parity with
                 // the behavior you get when the user does not declare the return type and with Json.Net at least at the top level.
-                var objectType = context.Object?.GetType() ?? context.ObjectType;
+                var objectType = context.Object?.GetType() ?? context.ObjectType ?? typeof(object);
```

Comment:
@ahsonkhan Why did this [behavior change](https://github.com/dotnet/runtime/pull/528/files#r358573412)? Seems random and unnecessary.

cc @pranavkm